### PR TITLE
Option to have absorbing BC for particles, regardless of field BC

### PR DESCRIPTION
**Pull request by MaxThevenet**
Ideally, we could set field and particle BC independently. This is currently not possible, and BC are mostly handled in AMReX. This PR is a first towards more independent BC for particles: right before the call to `MultiParticleContainer::Redistribute`, we call `MultiParticleContiner::ApplyBoundaryConditions`, which either does nothing or applies absorbing boundary conditions for particles (discards particles, i.e., set their `id` to -1).

To use absorbing BC for particles, run the simulation with
```
particles.absorbing_bc = 1
```

In the future, we could have something smarter, e.g.,
```
particles.boundary_conditions = absorbing absorbing periodic
```

### CODE REVIEW

**Comment by ax3l** [comment]
Thank you, this is great!

Just to avoid a user-facing API break in the future, don't you immediately want to add a `particles.boundary_conditions` option and just restrict the currently available values? :)
`reflecting` is another typical boundary condition that comes to mind to study shocks.

**Comment by ax3l** [comment]
openPMD files store the particle and field boundaries [here](https://github.com/ECP-WarpX/WarpX/blob/789f0da95e409b035cfffcacbb75dc847243e30a/Source/Diagnostics/WarpXOpenPMD.cpp#L693-L712), do you like to update the logic therein with the PR? :)
(I can also push on the PR to update it once I understand the logic change, see other questions.)

File: Docs/source/running_cpp/parameters.rst [hunk 1]
```diff
@@ -277,6 +277,9 @@ Particle initialization
 * ``particles.use_fdtd_nci_corr`` (`0` or `1`) optional (default `0`)
     Whether to activate the FDTD Numerical Cherenkov Instability corrector.
 
+* ``particles.absorbing_bc`` (`0` or `1`) optional (default `0`)
+    Whether to use absorbing boundary conditions for particles, regardless of the boundary conditions for the fields.
```
Comment (ax3l) [inline Docs/source/running_cpp/parameters.rst]: I think the 2nd half sentence might be confusing (or it confuses me). The default will be `0`, which means particles get wrapped around or continue propagating outside of the geometry bounds? Is there any leftover dependency of the behavior that depends on the field boundary setting?
```suggestion
    Whether to use absorbing boundary conditions for particles.
```

Comment (MaxThevenet) [reply Docs/source/running_cpp/parameters.rst]: When `particles.absorbing_bc = 0`, particle BC is determined by `geometry.is_periodic`:
- If `geometry.is_periodic = 1` exiting particles will be injected from the other side.
- If `geometry.is_periodic = 0` exiting particles are deleted.

Ideally particle BC would be independent on field BC, but this is out of the scope of this PR.

**Comment by MaxThevenet** [comment]
Thanks for you review @ax3l, the last commit should include fixes you suggested. Hopefully, the documentation is less confusing now, and still accurate.

**Review by ax3l — APPROVED** [review]
Great solution with the named options, together with the doc that addresses the prior ambiguity :+1: 

Thanks a lot! :sparkles:

**Pull request merged**